downstairs/src/lib.rs

```diff
@@ -1898,11 +2022,14 @@ impl Downstairs {
                     data,
                 }))
             }
-            IOop::WriteUnwritten { writes, .. } => {
+            IOop::WriteUnwritten { .. } => {
                 /*
                  * Any error from an IO should be intercepted here and passed
                  * back to the upstairs.
                  */
+                let IOop::WriteUnwritten { writes, .. } = job.work else {
```

Comment:
Why not change to `match job.work` above?

Reply:
Good catch – that was previously not possible due to the types in `IOop::ReadRequest`, but that was fixed by #1283 .  Fixed in b70e4f1f118f